src/permissions/PermissionCallableAllowedContractNativeTokenRecurringAllowance.sol

```diff
@@ -89,20 +91,24 @@ contract PermissionCallableAllowedContractNativeTokenRecurringAllowance is
 
             if (selector == IPermissionCallable.permissionedCall.selector) {
                 // check call target is the allowed contract
-                if (call.target != values.allowedContract) revert UserOperationLib.TargetNotAllowed();
+                if (call.target != values.allowedContract) revert UserOperationLib.TargetNotAllowed(call.target);
                 // assume PermissionManager already prevents account as target
             } else if (selector == IMagicSpend.withdraw.selector) {
+                // check call target is MagicSpend
+                if (call.target != magicSpend) revert UserOperationLib.TargetNotAllowed(call.target);
```

Comment:
Key logic change: only should be able to call this selector on magic spend contract.